scripts/azure/storage-manager.sh

```diff
@@ -0,0 +1,157 @@
+        log_info "resource group [${CK8S_ENVIRONMENT_NAME}-storage-resource-group] already exists" >&2
+        log_info "continue using this group ? (y/n)" >&2
+        read -r -n 1 cmdinput
+        case "$cmdinput" in
+        y|Y) return ;;
+        *) exit 0 ;;
+        esac
+    else
+        log_info "resource group [${CK8S_ENVIRONMENT_NAME}-storage-resource-group] does not exist, creating it now" >&2
+        az group create \
+            --name "$CK8S_ENVIRONMENT_NAME"-storage-resource-group \
+            --location "${AZURE_LOCATION}" --only-show-errors
+    fi
+}
+
+function create_storage_account() {
+
+    log_info "checking storage account availability" >&2
+    out=$(az storage account check-name --only-show-errors --name "${CK8S_ENVIRONMENT_NAME}"storageaccount)
+    ACCOUNT_AVAILABLE=$(echo "$out" | jq -r .nameAvailable)
+    REASON=$(echo "$out" | jq -r .reason)
+    case $ACCOUNT_AVAILABLE in
+        false)
+            if [ "$REASON" == "AccountNameInvalid" ]; then
+                log_info "Account name invalid, must be only contain numbers and lowercase letters"
+                exit 0
+            elif [ "$REASON" == "AlreadyExists" ]; then
+                log_info "storage account [${CK8S_ENVIRONMENT_NAME}storageaccount] already exists" >&2
+                log_info "contnue using this account ? (y/n)" >&2
+                read -r -n 1 cmdinput
+                case "$cmdinput" in
+                y) return ;;
+                n) exit 0 ;;
+                esac
+            fi
+            ;;
+        true)
+            log_info "creating storage account ${CK8S_ENVIRONMENT_NAME}storageaccount"
+            az storage account create \
+            --name "$CK8S_ENVIRONMENT_NAME"storageaccount \
+            --resource-group "$CK8S_ENVIRONMENT_NAME"-storage-resource-group \
+            --location swedencentral \
+            --sku Standard_RAGRS \
+            --kind StorageV2 \
+            --allow-blob-public-access false --only-show-errors
+            ;;
+    esac
+}
+
+function create_containers() {
+
+    CONTAINERS_LIST=$(az storage container list --account-name "$CK8S_ENVIRONMENT_NAME"storageaccount --query '[].name' --only-show-errors)
+
+    # shellcheck disable=SC2068
+    for container in ${CONTAINERS[@]}; do
+
+        log_info "checking status of container ${container}" >&2
+
+        CONTAINER_EXISTS=$(echo "$CONTAINERS_LIST" | jq --arg container "${container}" '. | index($container)')
+
+        if [ "$CONTAINER_EXISTS" != null ]; then
+            log_info "container ${container} already exists, do nothing" >&2
+        else
+            log_info "container ${container} does not exist, creating it now" >&2
+            az storage container create \
+                -n "$container" \
+                --account-name "$CK8S_ENVIRONMENT_NAME"storageaccount --only-show-errors
+        fi
+    done
+}
+
+function delete_all() {
+    az group delete --name "$CK8S_ENVIRONMENT_NAME"-storage-resource-group
```

Comment:
Consider making the resource group name a global variable since it's used throughout the script and if you update it at one place it could be easy to updated it in all places if you're not careful!